extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/TemplatePathBuildItem.java

```diff
@@ -1,28 +1,46 @@
 package io.quarkus.qute.deployment;
 
 import java.nio.file.Path;
+import java.util.Objects;
 
 import io.quarkus.builder.item.MultiBuildItem;
 
 /**
- * Represents a template path.
+ * Discovered template.
+ * <p>
+ * Templates backed by files located in a template root are discovered automatically. Furthermore, extensions can produce this
+ * build item in order to provide a template that is not backed by a file.
+ *
+ * @see TemplateRootBuildItem
  */
 public final class TemplatePathBuildItem extends MultiBuildItem {
 
     static final String TAGS = "tags/";
 
     private final String path;
-    private final Path fullPath;
     private final String content;
+    private final Path fullPath;
+    private final String extensionInfo;
+
+    public TemplatePathBuildItem(String path, String extensionInfo, String content) {
```

Comment:
@mkouba could you create a builder and add a bit of doc for the different fields?

Reply:
Sure, why not ;-)

Reply:
Done.